### PR TITLE
Add CardView component

### PR DESCRIPTION
**Pull request by michael-martin**
Adds a very simple UI component to display a Twitter card.

(And adds support for the spread operator)

### CODE REVIEW

File: src/components/CardView/CardView.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      <div className="card-block">
+        <h4 className="card-title">{title}</h4>
+        <p className="card-text">{text}</p>
+        {buttonEle}
+      </div>
+    </div>
+  );
+};
+
+CardView.PropTypes = {
```
Comment (vitkon) [inline src/components/CardView/CardView.js]: should it be `CardView.propTypes` with lowercase `p`?

File: src/components/CardView/CardView.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes } from 'react';
+
+const CardView = (props) => {
+  const { title, text, imageUrl, imageAlt, buttonUrl, buttonText } = props;
+
+  let imageEle;
+  if (imageUrl) {
+    imageEle = <img className="card-img-top" src={imageUrl} alt={imageAlt} />;
+  }
```
Comment (vitkon) [inline src/components/CardView/CardView.js:9]: can we use ternary here and make`imageEle` immutable (`const`)?

File: src/components/CardView/CardView.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  let buttonEle;
+  if (buttonUrl && buttonText) {
+    buttonEle = <a href={buttonUrl} className="btn btn-secondary" target="_blank">{buttonText}</a>;
+  }
```
Comment (vitkon) [inline src/components/CardView/CardView.js:20]: can we use ternary with button too?

File: src/components/CardView/CardView.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  return (
+    <div className="card">
+      {imageEle}
+      <div className="card-block">
+        <h4 className="card-title">{title}</h4>
+        <p className="card-text">{text}</p>
```
Comment (vitkon) [inline src/components/CardView/CardView.js:27]: can we follow BEM convention for classes?

e.g. `card__block`, `card__title` etc

File: src/components/CardView/CardView.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  const { title, text, imageUrl, imageAlt, buttonUrl, buttonText } = props;
+
+  let imageEle;
+  if (imageUrl) {
+    imageEle = <img className="card-img-top" src={imageUrl} alt={imageAlt} />;
+  }
+
+  let buttonEle;
+  if (buttonUrl && buttonText) {
+    buttonEle = <a href={buttonUrl} className="btn btn-secondary" target="_blank">{buttonText}</a>;
```
Comment (vitkon) [inline src/components/CardView/CardView.js]: can usage of link for a button be as accessible as using `<button>`?
Will you be able to disable it?

Comment (vitkon) [inline src/components/CardView/CardView.js]: Remember the _blank vulnerability?
Can we add `rel="noopener noreferrer"`?

https://medium.com/@jitbit/target-blank-the-most-underestimated-vulnerability-ever-96e328301f4c#.vls2com1w

File: src/components/CardView/CardView.js (hunk 1 above)
Comment (michael-martin) [inline src/components/CardView/CardView.js]: Woops, good catch!

File: src/components/CardView/CardView.js (hunk 2 above)
Comment (michael-martin) [inline src/components/CardView/CardView.js:9]: Breaks the line length rule :( (especially on the button. I need to put that on multiple lines with the extra param

File: src/components/CardView/CardView.js (hunk 4 above)
Comment (michael-martin) [inline src/components/CardView/CardView.js:27]: No need. CSS modules are locally scoped, so no need for BEM namespacing! :) 

EDIT - Just realised you meant the string class names. They're Bootstrap classes (Saves writing any new CSS for the sake of the demo! :) )

File: src/components/CardView/CardView.js (hunk 5 above)
Comment (michael-martin) [inline src/components/CardView/CardView.js]: Added the noopener, but I think it's best to leave the button as-is.

If we were building out a full app here, I'd consider it, but most of this code is intended to be deleted when the app is cloned. It is an example of building out features, but once you've built your own, I expect anything we were here functionality-wise to be removed, so keen to keep that as small as possible.

**Pull request merged**